### ISSUE TITLE
你好，之前我咨询过你。现在，我感觉你的代码有问题！

### ISSUE THREAD

**Issue by gengyanlei** [hfslyc/AdvSemiSeg#11]
@hfslyc 
设置梯度回传为False，这个位置不对。按照你的代码推理，对抗过程应该不能设置梯度为False。
 # train D
 # bring back requires_grad
 for param in model_D.parameters():
       param.requires_grad = True

# train with pred
pred = pred.detach()

您可以给出解释么？我在语义分割对抗方面是小白。

**Comment by hfslyc** [comment]
Hi, please explain more regarding your issue. I couldn't follow

**Comment by gengyanlei** [comment]
What I mean is that I'm doing semi-supervised training based on your code, but the effect of semi-supervised training is worse than supervised training. My training is a binary problem.

**Comment by gengyanlei** [comment]
What I mean is that I'm doing semi-supervised training based on your code, but the effect of semi-supervised training is worse than supervised training. My training is a binary problem.

My main function is similar to your main function, but the G network I use is U_Net.

**Comment by hfslyc** [comment]
Hi,

for a different dataset, you might want to experiment with different values of "lambda_adv_pred" to see whether the adversarial training make sense, then continue to experiment with the semi-supervised setting.

Since this is not a bug regarding the code, I'm closing this one for now.
